packages/trail-core/lib/index.js

```diff
@@ -87,16 +87,38 @@ class TrailsManager {
           ("when" >= ${from.toISO()} AND "when" <= ${to.toISO()})
     `
 
+    // Prepare the query to get Count
+    const sqlCount = SQL`
+      SELECT
+          count(*)
+        FROM trails
+        WHERE
+          ("when" >= ${from.toISO()} AND "when" <= ${to.toISO()})
+    `
+
     const op = caseInsensitive ? 'ILIKE' : 'LIKE'
-    if (who) sql.append(SQL([` AND who_id ${op} `])).append(SQL`${exactMatch ? who : '%' + who + '%'}`)
-    if (what) sql.append(SQL([` AND what_id ${op} `])).append(SQL`${exactMatch ? what : '%' + what + '%'}`)
-    if (subject) sql.append(SQL([` AND subject_id ${op} `])).append(SQL`${exactMatch ? subject : '%' + subject + '%'}`)
+    if (who) {
+      sql.append(SQL([` AND who_id ${op} `])).append(SQL`${exactMatch ? who : '%' + who + '%'}`)
+      sqlCount.append(SQL([` AND who_id ${op} `])).append(SQL`${exactMatch ? who : '%' + who + '%'}`)
+    }
+    if (what) {
+      sql.append(SQL([` AND what_id ${op} `])).append(SQL`${exactMatch ? what : '%' + what + '%'}`)
+      sqlCount.append(SQL([` AND what_id ${op} `])).append(SQL`${exactMatch ? what : '%' + what + '%'}`)
+    }
+    if (subject) {
+      sql.append(SQL([` AND subject_id ${op} `])).append(SQL`${exactMatch ? subject : '%' + subject + '%'}`)
+      sqlCount.append(SQL([` AND subject_id ${op} `])).append(SQL`${exactMatch ? subject : '%' + subject + '%'}`)
+    }
```

Comment:
this could use some cleanup to avoid repeating the query fragments multiple times

Reply:
will do ..

Reply:
this way ?
```
if (who) {
      const whoClause = SQL([` AND who_id ${op} `]).append(SQL`${exactMatch ? who : '%' + who + '%'}`)
      sql.append(whoClause)
      sqlCount.append(whoClause)
    }
```

Reply:
yes, that's an option. or concat all the sql together then append it to sql and sqlCount at the end. just avoid duplication basically